#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
IMPORTABLE EXTERNPROTO "../protos/Elisa3.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/Parquetry.proto"

WorldInfo {
  info [
    "The model of the Elisa-3 robot"
  ]
  title "Elisa-3"
}
Viewpoint {
  orientation -0.5283107492504255 0.5349937488102706 0.6592946541269595 1.94642296932938
  position -0.09983892793394554 -1.0661757276705532 3.8357752773211358
}
TexturedBackground {
}
TexturedBackgroundLight {
  luminosity 0.3
}
RectangleArena {
  floorSize 2 2
  floorAppearance Parquetry {
    type "light strip"
  }
}
Robot {
  children [
    Receiver {
      name "receiver_reset"
      channel 6
    }
    Emitter {
      name "emitter_RL"
      channel 4
    }
    Receiver {
      name "receiver_RL"
      channel 5
    }
    Emitter {
      name "emitter1"
      channel 1
    }
    Emitter {
      name "emitter2"
      channel 2
    }
    Emitter {
      name "emitter3"
      channel 3
    }
  ]
  name "supervisor"
  controller "supervisor"
  supervisor TRUE
}
DEF ADV1 Elisa3 {
  translation 0.49 -2.6378655862226877e-37 -0.010045440000000003
  rotation -1 0 0 1.783268467025218e-35
  name "adv1"
  controller "adv1"
  emitter_channel 7
}
DEF ADV2 Elisa3 {
  translation 0.59 -2.63787e-37 0
  rotation -1 0 0 1.783268467025218e-35
  name "adv2"
  controller "adv2"
  emitter_channel 8
  receiver_channel 2
}
DEF AGENT Elisa3 {
  translation 0.69 -2.63787e-37 0
  rotation -1 0 0 1.783268467025218e-35
  name "agent"
  controller "agent"
  emitter_channel 9
  receiver_channel 3
}
Robot {
  children [
    Receiver {
      name "receiver_vel_adv1"
      channel 7
    }
    Receiver {
      name "receiver_vel_adv2"
      channel 8
    }
    Receiver {
      name "receiver_vel_agent"
      channel 9
    }
    Emitter {
      name "emitter_reset"
      channel 6
    }
    Receiver {
      channel 4
    }
    Emitter {
      channel 5
    }
  ]
  name "RL supervisor"
  controller "RL_supervisor"
}
